From: <ohering@suse.de>
Date: Tue, 4 Sep 2012 17:54:13 -0700
Patch-mainline: 3.7-rc1
Subject: [PATCH] Drivers: hv: kvp: Copy the address family information
Git-commit: a500e0e73514ee2b838e439d3eb0750dda1da20f

This is part of the IP injection protocol in that the host expects this field
to reflect what addresses (address families) are currently bound to the
interface. The KVP daemon is currently collecting this information and sending
it to the kernel component. I had overlooked copying this and sending it
back to the host. This patch addresses this issue.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Reviewed-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/hv/hv_kvp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/hv/hv_kvp.c b/drivers/hv/hv_kvp.c
index d906050..ed50e9e 100644
--- a/drivers/hv/hv_kvp.c
+++ b/drivers/hv/hv_kvp.c
@@ -256,6 +256,8 @@ static int process_ob_ipinfo(void *in_msg, void *out_msg, int op)
 
 		out->kvp_ip_val.dhcp_enabled =
 			in->body.kvp_ip_val.dhcp_enabled;
+		out->kvp_ip_val.addr_family =
+			in->body.kvp_ip_val.addr_family;
 	}
 
 	return 0;
-- 
1.7.12.1

